UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -83,10 +94,23 @@ private void Update()
             }
         }
 
+        // if player has to slide then add sideways speed to make it go down
+        if (shouldSlide)
+        {
+            movementVector.x += (1f - hitNormal.y) * hitNormal.x * (speed - slideFriction);
+            movementVector.z += (1f - hitNormal.y) * hitNormal.z * (speed - slideFriction);
+        }
+        
+
         //Apply the result and move the character in space
         movementVector.y = verticalMovement;
         characterController.Move(movementVector * Time.deltaTime);
 
+        // check if the controller is grounded and above slope limit
+        // if player is grounded and above slope limit
+        // player has to slide
+        shouldSlide = !(Vector3.Angle(Vector3.up, hitNormal) <= characterController.slopeLimit);
```

Comment:
Maybe add some variables to clean this code part?
Adding more variables is a good way to increase legibility among other developers.

Reply:
Thank you. I tried to change it a little bit how does it look now? I am just a beginner so i have a lot to learn.

Reply:
Sometimes a good variable or method name can replace a lot of comments. Also, it indicates you care about legibility which is always good.